src/main/java/sparksoniq/io/json/JiqsItemParser.java

```diff
@@ -68,7 +68,12 @@ public static Item getItemFromObject(JsonIterator object, IteratorMetadata metad
                 {
                     return ItemFactory.getInstance().createDecimalItem(new BigDecimal(number));
                 }
-                return ItemFactory.getInstance().createIntegerItem(Integer.parseInt(number));
+                try {
+                	return ItemFactory.getInstance().createIntegerItem(Integer.parseInt(number));
+                } catch (NumberFormatException e)
+                {
+                	return ItemFactory.getInstance().createDecimalItem(new BigDecimal(number));
+                }
```

Comment:
This part reminded me of this comment: https://github.com/RumbleDB/rumble/pull/350#discussion_r343552206

I wonder if there could be an alternative to implement this without the exception and if this would be beneficial for performance.

Reply:
That's a very good point. Unfortunately, I haven't found an easy way to test for parsability as an int.

We could find a middle ground by testing for the length of the string before trying to cast: if it is at least 12 characters long, we can directly create a decimal.

This restricts the exception throwing to strings of 10 chars (no sign) or 11 (with sign), which is a rarer case...

What do you think?

Reply:
I like the idea of shrinking the surface where exceptions can be triggered but I'm not sure that I entirely grasp which exceptions we're trying to cover and optimize against here.

I see that double and decimal types which contain (e/E) and (.) respectively are already covered.
I'm assuming that non-number inputs are not arriving to this stage of execution (if they did, we would have other exceptions with parseDouble for example).
So the case that i can think about at the moment are the values for which the integer type is not large enough to store the value (assuming this causes a NumberFormatException). If so, firstly, why is number 12 chosen here? Secondly, are there any other triggers of exceptions that I'm missing?

Reply:
No, only that -- this is a JSON parser that already detected that this is a number, so not much else can happen at that point.

So we want to use Decimal if 32 bits are exceeded (which happened with a dataset I am using). The NumberFormatException will happen if the 32 bits are exceeded.

We need at least 12 to be sure it does not fit because 11 could be a negative 32-bit integer: "-1000000000". We could also have an even more fine-grained test looking for a sign and the first digit (if we have 10 digits and the first one is above 5, we can also avoid the exception throwing).